app/controllers/claimToAdjustPoa/EnterPoAAmountController.scala

```diff
@@ -66,41 +70,55 @@ class EnterPoAAmountController @Inject()(val authorisedFunctions: AuthorisedFunc
         }
     }
 
-  def submit(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+  def submit(isAgent: Boolean, mode: Mode): Action[AnyContent] = auth.authenticatedAction(isAgent) {
     implicit request =>
       ifAdjustPoaIsEnabled(isAgent) {
         claimToAdjustService.getEnterPoAAmountViewModel(Nino(request.nino)).flatMap {
           case Right(viewModel) =>
-            handleForm(viewModel, isAgent)
+            handleForm(viewModel, isAgent, mode)
           case Left(ex) =>
             Logger("application").error(s"Error while retrieving charge history details : ${ex.getMessage} - ${ex.getCause}")
             Future.successful(showInternalServerError(isAgent))
         }
       }
   }
 
-  def handleForm(viewModel: PoAAmountViewModel, isAgent: Boolean)(implicit user: MtdItUser[_]) = {
+  def handleForm(viewModel: PoAAmountViewModel, isAgent: Boolean, mode: Mode)(implicit user: MtdItUser[_]) = {
     EnterPoaAmountForm.checkValueConstraints(EnterPoaAmountForm.form.bindFromRequest(), viewModel.totalAmountOne, viewModel.relevantAmountOne).fold(
       formWithErrors =>
-        Future.successful(BadRequest(view(formWithErrors, viewModel, isAgent, controllers.claimToAdjustPoa.routes.EnterPoAAmountController.submit(isAgent)))),
+        Future.successful(BadRequest(view(formWithErrors, viewModel, isAgent, controllers.claimToAdjustPoa.routes.EnterPoAAmountController.submit(isAgent, mode)))),
       validForm =>
         sessionService.setNewPoAAmount(validForm.amount).flatMap {
           case Left(ex) => Logger("application").error(s"Error while setting mongo data : ${ex.getMessage} - ${ex.getCause}")
             Future.successful(showInternalServerError(isAgent))
-          case Right(_) => getRedirect(viewModel, validForm.amount, isAgent)
+          case Right(_) => getRedirect(viewModel, validForm.amount, isAgent, mode)
         }
     )
   }
 
-  def getRedirect(viewModel: PoAAmountViewModel, newPoaAmount: BigDecimal, isAgent: Boolean)(implicit user: MtdItUser[_]): Future[Result] = {
-    (viewModel.totalAmountLessThanPoa, newPoaAmount > viewModel.totalAmountOne) match {
+  def getRedirect(viewModel: PoAAmountViewModel, newPoaAmount: BigDecimal, isAgent: Boolean, mode: Mode)(implicit user: MtdItUser[_]): Future[Result] = {
+    val canIncrease = viewModel.totalAmountLessThanPoa
```

Comment:
* canIncrease and hasIncreased can be move into viewModel, as these are reflection of viewModel properties;